src/phy/mod.rs

```diff
@@ -265,3 +265,8 @@ pub trait TxToken {
     fn consume<R, F>(self, timestamp: Instant, len: usize, f: F) -> Result<R>
         where F: FnOnce(&mut [u8]) -> Result<R>;
 }
+
+/// Ethernet header length to be added to the MTU, since the header is not counted into MTU
+/// 6 bytes src addr, 6 bytes dst addr, 2 bytes EthType,
+/// the optionbal 4 bytes of IEEE 802.1Q Header are not accounted for
+pub const ETHERNET_HAEDER_MAX_LEN: usize = 14;
```

Comment:
I think it would be better to use 18 here as correct maximal length because otherwise you could just use `EthernetFrame::<&[u8]>::header_len()` which is 14.

When using 18, this has to be changed as well:
```
diff --git a/src/iface/ethernet.rs b/src/iface/ethernet.rs
index e44abb4..e3fd029 100644
--- a/src/iface/ethernet.rs
+++ b/src/iface/ethernet.rs
@@ -388,7 +388,7 @@ impl<'b, 'c, 'e, DeviceT> Interface<'b, 'c, 'e, DeviceT>
 
     fn socket_egress(&mut self, sockets: &mut SocketSet, timestamp: Instant) -> Result<bool> {
         let mut caps = self.device.capabilities();
-        caps.max_transmission_unit -= EthernetFrame::<&[u8]>::header_len();
+        caps.max_transmission_unit -= ETHERNET_HAEDER_MAX_LEN;
 
         let mut emitted_any = false;
         for mut socket in sockets.iter_mut() {
@@ -1122,7 +1122,7 @@ impl<'b, 'c, 'e> InterfaceInner<'b, 'c, 'e> {
                     // I'm really not happy about this "solution" but I don't know what else to do.
                     if let Some(max_burst_size) = caps.max_burst_size {
                         let mut max_segment_size = caps.max_transmission_unit;
-                        max_segment_size -= EthernetFrame::<&[u8]>::header_len();
+                        max_segment_size -= ETHERNET_HAEDER_MAX_LEN;
                         max_segment_size -= ip_repr.buffer_len();
                         max_segment_size -= tcp_repr.header_len();
```